### PR TITLE
AC-1935 - Organize tracking logic and enhance submit event for testing

### PR DESCRIPTION
**Pull request by nstanard**
### What Changed

- Moved the new Form component into a subfolder called tracking. Any future components related to tracking activity should be put here too. (Higher order tracking components that wrap the main components - to keep all tracking logic in one central location)
- Logging the form information but only for non production ENVs if the local debug is set.

### How To Test

- Check that the Form component is still imported correctly.
- Check that the FieldSet component is still imported correctly.
- Check that the logging in the Form submit event is only happening on non production envs (for now.. that part is temporary until we actually hook up the tracking call and pass info into it per Form instance.)

### CODE REVIEW

File: src/components/datePicker/ManualEntryForm.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ import moment from 'moment';
 import _ from 'lodash';
 import { ArrowForward } from '@sparkpost/matchbox-icons';
 import { Grid, TextField } from 'src/components/matchbox';
-import { Form } from 'src/components/form';
+import { Form } from 'src/components/tracking/form';
```
Comment (nstanard) [inline src/components/datePicker/ManualEntryForm.js:6]: `http://localhost:3100/lists/suppressions` -> path to test

(DatePicker and this file are not currently firing the main tracking/form submit event.)

Comment (shrirupa) [reply src/components/datePicker/ManualEntryForm.js:6]: AC-1937

File: src/components/datePicker/ManualEntryFormV2.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ import React, { useReducer, useEffect, useCallback } from 'react';
 import moment from 'moment';
 import _ from 'lodash';
 import { Grid, TextField } from 'src/components/matchbox';
-import { Form } from 'src/components/form';
+import { Form } from 'src/components/tracking/form';
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/components/datePicker/ManualEntryFormV2.js:5]: AC-1937

**Review by shrirupa — APPROVED** [review]
looks good to me!

**Comment by nstanard** [comment]
GH was having issues, this PR got merged but isnt' showing as such. Here is the squash in master history for reference: 
https://github.com/SparkPost/2web2ui/commit/bb883836d922ed04892fb771203221ca8622f7cc

Closing this PR.

**Pull request closed without merging**